CMakeLists.txt

```diff
@@ -152,8 +152,23 @@ set_target_properties(fbgemm_generic fbgemm_avx2 fbgemm_avx512 PROPERTIES
       CXX_VISIBILITY_PRESET hidden)
 
 if(MSVC)
+  if(FBGEMM_LIBRARY_TYPE STREQUAL "static")
+     target_compile_definitions(fbgemm_generic PRIVATE ASMJIT_STATIC)
+     target_compile_definitions(fbgemm_avx2 PRIVATE ASMJIT_STATIC)
+     target_compile_definitions(fbgemm_avx512 PRIVATE ASMJIT_STATIC)
+     foreach(flag_var
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+       if(${flag_var} MATCHES "/MD")
+         string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+       endif(${flag_var} MATCHES "/MD")
+     endforeach(flag_var)
+  endif()
   target_compile_options(fbgemm_avx2 PRIVATE "/arch:AVX2")
   target_compile_options(fbgemm_avx512 PRIVATE "/arch:AVX512")
+
+  set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:MSVCRT /ignore:4049")
```

Comment:
What's 4049 warning? Can we add a comment on it?

Reply:
I will add comments for these changes.
We can ignore this one. I forgot to remove it.